src/views/HousingLottery/index.jsx

```diff
@@ -1,7 +1,99 @@
-  return <PreferredHall />;
-};
+  const [message, setMessage] = useState('');
+  const [morningOrNight, setMorningOrNight] = useState('');
+  const [loudOrQuiet, setLoudOrQuiet] = useState('');
+  const [preferences, setPreferences] = useState([]); 
+  
+
+  const handleChange = (event) => {
+    setMessage(event.target.value);
+  };
+
+  const handleMorningOrNightChange = (event) => {
+    setMorningOrNight(event.target.value);
+  };
+
+  const handleLoudOrQuietChange = (event) => {
+    setLoudOrQuiet(event.target.value);
+  };
+
+  const handlePreferenceChange = (event) => {
+    // Update preferences based on the selected radio button
+    const newPreferences = event.target.value;
+    setPreferences((prevPreferences) => {
+      if (prevPreferences.includes(newPreference)) {
+        // Remove the preference if it's already in the array
+        return prevPreferences.filter((pref) => pref !== newPreference);
+      } else {
+        // Add the preference if it's not in the array
+        return [...prevPreferences, newPreference];
+      }
+    });
+  };
+  const handleClick = async () => {
+    // You can access message, morningOrNight, and loudOrQuiet to submit to your housing service
+    await housingService.addRoommate({ message, preferences});
+  };
+
+  return (
+    <Grid container justifyContent="center">
+      <Grid item xs={12} lg={6} style={{ maxWidth: '500px' }}>
+        <Card>
+          <CardHeader title="Preferences" className={styles.apartment_card_header} />
+          <CardContent>
+            <div>
+              <label>Are you a night owl or a morning bird?</label>
+              <RadioGroup
+                aria-label="morning-or-night"
+                name="morning-or-night"
+                value={morningOrNight}
+                onChange={handleMorningOrNightChange}
+              >
+                <FormControlLabel value="night-owl" control={<Radio />} label="Night Owl" />
+                <FormControlLabel value="morning-bird" control={<Radio />} label="Morning Bird" />
+              </RadioGroup>
+            </div>
+
+            <div>
+              <label>Do you consider yourself quiet or loud in the dorm?</label>
+              <RadioGroup
+                aria-label="loud-or-quiet"
+                name="loud-or-quiet"
+                value={loudOrQuiet}
+                onChange={handleLoudOrQuietChange}
+              >
+                <FormControlLabel value="quiet" control={<Radio />} label="Quiet" />
+                <FormControlLabel value="loud" control={<Radio />} label="Loud" />
+              </RadioGroup>
+            </div>
+          </CardContent>
+          <Grid container justifyContent="flex-end" style={{ padding: '0 16px 16px 0' }}>
+            <Grid item>
+              <Button variant="contained" onClick={handleClick}>
+                Submit
+              </Button>
+            </Grid>
+          </Grid>
+        </Card>
+      </Grid>
+    </Grid>
+  );
+  };
```

Comment:
This looks nice! However, can you put these codes into another file, such as Preference/index.jsx, and only have <Preference /> in this file? We will have more and more code in this main index.jsx file, so we shouldn't put all codes in here.

Reply:
sure thing. I should place these in the other file instead piling in the index.jsx file